indico/modules/rb_new/controllers/backend/admin.py

```diff
@@ -252,3 +259,87 @@ def _check_conflict(self, name):
             query = query.filter(RoomAttribute.id != self.attribute.id)
         if query.has_rows():
             abort(422, messages={'name': [_('Name must be unique')]})
+
+
+class RHRoomAdminBase(RHRoomBookingBase):
+    def _process_args(self):
+        self.room = Room.get_one(request.view_args['room_id'])
+
+
+class RHRoomAttributesAll(RHRoomBookingBase):
+    def _process_args(self):
+        attributes = RoomAttribute.query.all()
+        return jsonify(attributes_schema.dump(attributes).data)
+
+
+class RHRoomAttributes(RHRoomAdminBase):
+    def _process(self):
+        return jsonify(room_attribute_values_schema.dump(self.room.attributes).data)
+
+
+class RHRoomAttributesUpdate(RHRoomAdminBase):
+    @use_kwargs({'attributes': fields.Nested({'title': fields.Str(),
+                                              'value': fields.Str(),
+                                              'name': fields.Str()}, many=True)})
+    def _process(self, attributes):
+        update_room_attributes(self.room, attributes)
+        return jsonify(room_attribute_values_schema.dump(self.room.attributes).data)
+
+
+class RHRoomAvailability(RHRoomAdminBase):
+    def _process(self):
+        return jsonify(nonbookable_periods=nonbookable_periods_schema.dump(self.room.nonbookable_periods, many=True).data,
+                       bookable_hours=bookable_hours_schema.dump(self.room.bookable_hours, many=True).data)
+
+
+class RHRoomAvailabilityUpdate(RHRoomAdminBase):
+    @use_args({'bookable_hours': fields.Nested({'start_time': fields.Time(),
+                                                'end_time': fields.Time()}, many=True),
+               'nonbookable_periods': fields.Nested({'start_dt': fields.Date(),
+                                                     'end_dt': fields.Date()}, many=True)})
+    def _process(self, args):
+        if 'bookable_hours' in args:
+            self._check_invalid_times(args)
+        update_room_availability(self.room, args)
+        return jsonify({'nonbookable_periods': nonbookable_periods_schema.dump(self.room.nonbookable_periods, many=True).data,
+                       'bookable_hours': bookable_hours_schema.dump(self.room.bookable_hours, many=True).data})
```

Comment:
indentation, and you can pass kwargs to jsonify(), no need to wrap it in a dict yourself